Assets/Scripts/Managers/DeliveryManager.cs

```diff
@@ -38,37 +38,49 @@ public bool DeliverMeat(List<Constants.Meats> requiredMeats, int maxBoxesToDeliv
         // Set number of boxes to be delivered to be equal to required meats
         // ie. I need one box of pepperoni, but for whatever reason, I pass in maxBoxesToDeliver as 0
         // I want maxBoxesToDeliver to be 1 so I still get the pepperoni box
-        if ( maxBoxesToDeliver < requiredMeats.Count )
+        int totalBoxesRequired = GetTotalBoxesCount( requiredMeats );
+        if ( maxBoxesToDeliver < totalBoxesRequired )
         {
-            maxBoxesToDeliver = requiredMeats.Count;
+            maxBoxesToDeliver = totalBoxesRequired;
         }
         
         // Populate list up to maxBoxes to deliver
-        List<Constants.Meats> meatBoxes = PopulateMeatList(requiredMeats, maxBoxesToDeliver);
+        List<Constants.Meats> meatBoxes = PopulateMeatList(requiredMeats, maxBoxesToDeliver, totalBoxesRequired);
 
         // Tell table to create and deliver them
         // Returns true if we did deliver the boxes
         return DeliveryTable.dtInstance.Deliver(meatBoxes);
     }
 
     // Create list of meats
-    private List<Constants.Meats> PopulateMeatList(List<Constants.Meats> requiredMeats, int boxesToDeliver)
+    private List<Constants.Meats> PopulateMeatList(List<PizzaOrder.MeatItem> requiredMeats, int boxesToDeliver, int totalBoxesRequired)
     {
+        List<Constants.Meats> finalMeats = new List<Constants.Meats>();
+
         // Copy over required meats
-        List<Constants.Meats> finalMeats = new List<Constants.Meats>(requiredMeats);
+        foreach ( PizzaOrder.MeatItem meatItem in requiredMeats )
+        {
+            for ( int i = 0; i < meatItem.numBoxes; i++ )
+            {
+                finalMeats.Add(meatItem.meatType);
+            }
+        }
 
         // Get random meats
-        // TODO: do we want to exclude required meats from sampling?
-        int initialCount = requiredMeats.Count;
         // Need to fill in with random meat up to boxes to deliver
-        for ( int i = 0; i < boxesToDeliver - initialCount; i++ )
+        if ( boxesToDeliver - totalBoxesRequired > 0 )
         {
-            int meatIdx = Random.Range(0, meatTypes.Count);
-            Constants.Meats meatType = meatTypes[meatIdx];
+            List<Constants.Meats> uniqueMeats = CreateUniqueMeatList(requiredMeats);
```

Comment:
Could be replaced with
```
HashSet<Constants.Meats> uniqueMeats = Enum.GetValues(typeof(Constants.Meats)).Cast<Constants.Meats>().ToHashSet();
uniqueMeats.ExceptWith(requiredMeats);
```
https://stackoverflow.com/questions/3816718/how-to-get-an-array-of-all-enum-values-in-c
Make sure you're `using System.Linq`!